src/query/v2/plan/operator_distributed.cpp

```diff
@@ -0,0 +1,473 @@
+    return std::make_optional(db->Vertices(view_, label_));
+  };
+  return MakeUniqueCursorPtr<ScanAllCursor<decltype(vertices)>>(mem, output_symbol_, input_->MakeCursor(mem),
+                                                                std::move(vertices), "ScanAllByLabel");
+}
+
+ScanAllByLabelPropertyValue::ScanAllByLabelPropertyValue(const std::shared_ptr<LogicalOperator> &input,
+                                                         Symbol output_symbol, storage::v3::LabelId label,
+                                                         storage::v3::PropertyId property,
+                                                         const std::string &property_name, Expression *expression,
+                                                         storage::v3::View view)
+    : ScanAll(input, output_symbol, view),
+      label_(label),
+      property_(property),
+      property_name_(property_name),
+      expression_(expression) {
+  DMG_ASSERT(expression, "Expression is not optional.");
+}
+
+ACCEPT_WITH_INPUT(ScanAllByLabelPropertyValue)
+
+UniqueCursorPtr ScanAllByLabelPropertyValue::MakeCursor(utils::MemoryResource *mem) const {
+  EventCounter::IncrementCounter(EventCounter::ScanAllByLabelPropertyValueOperator);
+
+  auto vertices =
+      [this](Frames &frames, ExecutionContext &context) -> std::optional<decltype(context.db_accessor->Vertices(
+                                                            view_, label_, property_, storage::v3::PropertyValue()))> {
+    MG_ASSERT(!frames.empty());
+    auto &frame = *frames[0];  // #NoCommit double check w.r.t ExpressionEvaluator, not sure this is correct.
+    auto *db = context.db_accessor;
+    ExpressionEvaluator evaluator(&frame, context.symbol_table, context.evaluation_context, context.db_accessor, view_);
+    auto value = expression_->Accept(evaluator);
+    if (value.IsNull()) {
+      return std::nullopt;
+    }
+    if (!value.IsPropertyValue()) {
+      throw QueryRuntimeException("'{}' cannot be used as a property value.", value.type());
+    }
+    return std::make_optional(db->Vertices(view_, label_, property_, storage::v3::PropertyValue(value)));
+  };
+  return MakeUniqueCursorPtr<ScanAllCursor<decltype(vertices)>>(mem, output_symbol_, input_->MakeCursor(mem),
+                                                                std::move(vertices), "ScanAllByLabelPropertyValue");
+}
+
+ScanAllById::ScanAllById(const std::shared_ptr<LogicalOperator> &input, Symbol output_symbol, Expression *expression,
+                         storage::v3::View view)
+    : ScanAll(input, output_symbol, view), expression_(expression) {
+  MG_ASSERT(expression);
+}
+
+ACCEPT_WITH_INPUT(ScanAllById)
+
+UniqueCursorPtr ScanAllById::MakeCursor(utils::MemoryResource *mem) const {
+  EventCounter::IncrementCounter(EventCounter::ScanAllByIdOperator);
+
+  auto vertices = [this](Frames &frames, ExecutionContext &context) -> std::optional<std::vector<VertexAccessor>> {
```

Comment:
Double check implementation.
If not correct, just remove as we don't need it right now.